contracts/carbon/Strategies.sol

```diff
@@ -435,7 +435,7 @@ abstract contract Strategies is Initializable {
             }
 
             sourceOrder.y += sourceAmount;
-            if (sourceOrder.z < sourceOrder.y) {
+            if (sourceOrder.z < sourceOrder.y && sourceOrder.A != 0) {
```

Comment:
How much gas does it save? This seems like a non-trivial change which will require a careful review.

Reply:
I agree.

It can save gas for the specific case of a trader selling his/her liquidity to a constant-rate order.
Potentially, 5k gas on average, for updating a slot from non-zero to non-zero (increasing the `z` value).

This will be particularly noticeable when trading with a single-order strategy, since the trader will be buying from that single order, hence - selling to the other order, which is a constant-rate order (of rate 0).

And it will actually happen on every trade on that irreversible strategy, until it is depleted (i.e., trader liquidity added to the invalid order on every trade, and the `z` value of the invalid order is updated on every trade).